### ISSUE TITLE
[BUG] Unconfigurable cutoff of “{{char}}:” at the beginning of a reply

### ISSUE THREAD

**Issue by pivlarek** [SillyTavern/SillyTavern#2666]
### Environment

🪟 Windows

### System

Yandex Browser 24

### Version

last staging

### Describe the problem

The formatting in my preset is based on the fact that each character should show the prefix `name:`. Sometimes (but not always) it happens that a message starts with the prefix `{{char}}:`, but this prefix is cut off despite having `Show {{char}}: in responses` turned on.

As I see in `script.js` in `cleanUpMessage`  function there is a cut out of `{{char}}:` or `{{user}}:`, which isn't controlled by the `allow_name1_display` and `allow_name2_display` settings.

```
const nameToTrim2 = isImpersonate ? name1 : name2;

if (getMessage.startsWith(nameToTrim2 + ':')) {
    getMessage = getMessage.replace(nameToTrim2 + ':', '');
    getMessage = getMessage.trimStart();
}
```

It seems logical to me that such a cutout should be controlled by the settings `Show {{char}}: in responses` and `Show {{user}}: in responses`. Or for example turn off when `Character Names Behavior` is `None`. But not be unconditional.

### Please tick the boxes

- [X] I have explained the issue clearly, and I included all relevant info
- [X] I have checked that this [issue hasn't already been raised](https://github.com/SillyTavern/SillyTavern/issues?q=is%3Aissue)
- [X] I have checked the [docs](https://docs.sillytavern.app/) ![important](https://img.shields.io/badge/Important!-F6094E)

**Comment by Wolfsblvt** [comment]
The code you mentioned is kind of part of the stopping strings handling. `name2` in that context is the users' name. This should prevent the AI from impersonating you inside the replies.

It should not be related to actual `{{char}}:` parts being cut off.

**Comment by pivlarek** [comment]
> The code you mentioned is kind of part of the stopping strings handling. `name2` in that context is the users' name. This should prevent the AI from impersonating you inside the replies.
> 
> It should not be related to actual `{{char}}:` parts being cut off.

At least `cleanUpMessage` is executed after the message is generated/streaming, and `name2` here is definitely `{{char}}`. As soon as I comment out the above code, the prefix shows up as it should.

**Comment by pivlarek** [comment]
> The code you mentioned is kind of part of the stopping strings handling. `name2` in that context is the users' name. This should prevent the AI from impersonating you inside the replies.
> 
> It should not be related to actual `{{char}}:` parts being cut off.

As far as I understand you mention this part of the code.

```
let nameToTrim = isImpersonate ? name2 : name1;

if (isImpersonate) {
    nameToTrim = power_user.allow_name2_display ? '' : name2;
}
else {
    nameToTrim = power_user.allow_name1_display ? '' : name1;
}

if (nameToTrim && getMessage.indexOf(`${nameToTrim}:`) == 0) {
    getMessage = getMessage.substring(0, getMessage.indexOf(`${nameToTrim}:`));
}
if (nameToTrim && getMessage.indexOf(`\n${nameToTrim}:`) >= 0) {
    getMessage = getMessage.substring(0, getMessage.indexOf(`\n${nameToTrim}:`));
}
```

But there is another trim section with `nameToTrim2` below in `cleanUpMessage` function that I mentioned in OP.

**Comment by Cohee1207** [comment]
PR is up. Test it out and let me know if that fixes it #2669
Please be aware that not having prefixes trimmed out from replies often leads to names being repeatedly stacked by the LLM. Be careful with that.

**Comment by pivlarek** [comment]
Yes, the bug is fixed. Thank you.
